src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -971,6 +973,113 @@ namespace
         }
     };
 
+    struct PlasticClosure
+    {
+        struct Params
+        {
+            OSL::ustring    dist;
+            OSL::Vec3       N;
+            OSL::Color3     specular_reflectance;
+            float           specular_reflectance_multiplier;
+            float           roughness;
+            float           highlight_falloff;
+            float           ior;
+            OSL::Color3     diffuse_reflectance;
+            float           diffuse_reflectance_multiplier;
+            float           internal_scattering;
+        };
+
+        static const char* name()
+        {
+            return "as_plastic";
+        }
+
+        static ClosureID id()
+        {
+            return PlasticID;
+        }
+
+        static int modes()
+        {
+            return ScatteringMode::Diffuse | ScatteringMode::Glossy | ScatteringMode::Specular;
```

Comment:
The specular flag corresponds to perfect or nearly perfect specular scattering. Does plastic has that?

Reply:
@dictoon i think so
https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/renderer/modeling/bsdf/plasticbrdf.cpp#L208

Reply:
Ok, good point.